google/cloud/spanner/internal/session.h

```diff
@@ -43,12 +46,16 @@ class Session {
 
   std::string const& session_name() const { return session_name_; }
 
+  void set_bad() { is_bad_.store(true, std::memory_order_release); }
```

Comment:
I believe we can relax these ordering constraints to `memory_order_relaxed`.  There is no state (other than `is_bad_` itself) that the `set_bad()` caller communicates to the `is_bad()` caller, so memory ordering is irrelevant.

Reply:
Done.